specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/applications.json

```diff
@@ -0,0 +1,312 @@
+        "operationId": "Application_Update",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          },
+          {
+            "$ref": "#/parameters/RuleId"
+          },
+          {
+            "$ref": "#/parameters/ApplicationId"
+          },
+          {
+            "$ref": "#/parameters/applicationInBody"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Application"
+            }
+          },
+          "default": {
+            "description": "Error response that describes why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Security/applications": {
+      "get": {
+        "x-ms-examples": {
+          "Get a list of applications": {
+            "$ref": "./examples/Applications/ListApplicationsByEnvironment_example.json"
+          }
+        },
+        "tags": [
+          "Application"
+        ],
+        "description": "Get a list of all relevant applications over a scope",
+        "operationId": "Application_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ApplicationList"
+            }
+          },
+          "default": {
+            "description": "Error response that describes why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Security/applicationMappingRule/{ruleId}/applications": {
+      "get": {
+        "x-ms-examples": {
+          "Get list of  applications": {
+            "$ref": "./examples/Applications/ListApplicationsByRuleId_example.json"
+          }
+        },
+        "tags": [
+          "Application"
+        ],
+        "description": "Get a list of all relevant applications over a rule Id.",
+        "operationId": "Application_ListByRuleId",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/RuleId"
```

Comment:
The parameters:ruleId,scope should be kept in the same order as they present in the path.

refs: https://github.com/Azure/azure-openapi-validator/blob/main/docs/parameter-order.md